astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -42,24 +41,24 @@ def assert_table_equal(t1, t2, check_meta=False, rtol=1.0e-15, atol=1.0e-300):
     Test equality of all columns in a table, with stricter tolerances for
     float columns than the np.allclose default.
     """
-    assert_equal(len(t1), len(t2))
-    assert_equal(t1.colnames, t2.colnames)
+    npt.assert_equal(len(t1), len(t2))
+    npt.assert_equal(t1.colnames, t2.colnames)
     if check_meta:
-        assert_equal(t1.meta, t2.meta)
+        npt.assert_equal(t1.meta, t2.meta)
```

Comment:
This is a bit trickier, as meta is neither a sequence nor a scalar, but typically a dict-like structure.
But here the numpy function should still provide better diagnostics:
```python
>>> assert t1.meta == t2.meta
AssertionError
>>> assert_equal(t1.meta, t2.meta)
AssertionError: 
Items are not equal:
key='value'
key='CLIPRANGE'
key='keywords'

 ACTUAL: '2.5'
 DESIRED: '2.0'
```

Reply:
The tracebacks the CPython interpreter produces can be very different from the tracebacks `pytest` produces:
```
============================================== FAILURES ===============================================
__________________________________________ test_example_good __________________________________________

    def test_example_good():
>       assert t1.meta == t2.meta
E       AssertionError: assert {'CLIPRANGE':... 'value': 2.5} == {'CLIPRANGE':... 'value': 2.0}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'value': 2.5} != {'value': 2.0}
E         Use -v to get more diff

example.py:11: AssertionError
__________________________________________ test_example_bad ___________________________________________

    def test_example_bad():
>       np.testing.assert_equal(t1.meta, t2.meta)
E       AssertionError: 
E       Items are not equal:
E       key='value'
E       
E        ACTUAL: 2.5
E        DESIRED: 2.0

example.py:15: AssertionError

```

Reply:
Yes, the pytest version might be a bit more useful here still. Would you suggest to leave numpy's `assert_equal` to array-like objects only and use `assert ...==` for everything else?

Reply:
When in doubt, make the test fail and see what the `pytest` traceback looks like. But for this pull request the reasonable thing to do would be to replace `assert_equal()` from `commons.py` with `assert ... == ...`, not `np.testing.assert_equal()`. `commons.assert_equal(x, y)` is basically an alias for `assert x == y` anyways, so that replacement will not make any potential tracebacks worse.

Reply:
@dhomeier  Should I start working on replacing the `npt.assert_equal` back to `assert x == y`.

Reply:
We have discussed in the meantime that `npt` is not really a standard alias used by astropy, so 
1. those imports should be changed to `numpy.testing import assert_allclose, assert_equal` in general
2. for comparing anything that is not an _array-like collection of numerical values_ (i.e. basically anything that can be converted to a standard numerical-dtype `ndarray`), per the suggestions above, replace `assert_equal` with `assert x == y`.
Thanks!

Reply:
Sure, will start my work as mentioned

Reply:
Just to clarify, we want
```python
assert_equal(arr_x, arr_y)
assert obj_x == obj_y
```